### ISSUE TITLE
'Convert to local function' isn't available at top level

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#83143]
**Version Used**:
VS 18.4.3 10f06335

**Steps to Reproduce**:
```cs
EventHandler handler = (sender, e) => Console.WriteLine("Hello World");
```

**Diagnostic Id:**
IDE0039: Use local function

**Expected Behavior**:
'Convert to local function' diagnostic is reported and code fix produces:
```cs
static void handler(object? sender, EventArgs e) => Console.WriteLine("Hello World");
```

**Actual Behavior**:
No diagnotic is reported
